pysph/base/tests/test_particle_array.py

```diff
@@ -34,7 +34,8 @@ class ParticleArrayTest(object):
 
     def test_constructor(self):
         # Default constructor test.
-        p = particle_array.ParticleArray(name='test_particle_array')
+        p = particle_array.ParticleArray(name='test_particle_array',
+                                         backend=self.backend)
```

Comment:
Is the backend compulsory now?  If so this API is not good, the backend ought to be detected automatically right?  Or is this just for the tests since we change the backend?